### PR TITLE
docs: clarify secrets documentation.

### PR DESCRIPTION
**Pull request by BenjaminMummery**
[ORQSDK-899]

# The problem

Some of the language in the `secrets` tutorial is unclear.

# This PR's solution

- “you may want to run a quantum circuit” → “you may want to run a quantum circuit on real quantum hardware”
- “authorization flow by following the step 1 from” → “authorization flow by following step 1 from”
- “If omitted, personal workspace will be used. You can get workspace_id from the workspace view in the web Orquestra Portal or by using list_workspace API call.” → “If omitted, your personal workspace will be used. You can get workspace_id from the workspace view in Orquestra Portal or by using `sdk.list_workspaces()`."
- Adds prerequisites

# Checklist

_Check that this PR satisfies the following items:_

- [X] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [X] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [X] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [X] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

[ORQSDK-899]: https://zapatacomputing.atlassian.net/browse/ORQSDK-899?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by alexjuda — CHANGES_REQUESTED** [review]

**Review by alexjuda — APPROVED** [review]
Oops, misclick. I left one non-blocking suggestion

File: docs/tutorials/secrets.rst [hunk 1]
```diff
@@ -29,11 +37,16 @@ Then, you can read and set secrets within a script or a Python REPL:
     :dedent: 8
 
 
-``name`` is the secret identifier. It can be any string you like, as long as it fits in the length limit. This identifier is subsequently used to access the secret. Secrets are scoped to a single user account, so there's no risk you're overriding somebody else's entry.
+``name`` is the secret identifier.
+It can be any string you like, as long as it fits in the length limit.
```
[the remark this answers is not in the document]
Comment (roryschadler) [reply docs/tutorials/secrets.rst:41]: Portal uses this regex for validation of secret names: `/^[a-zA-Z0-9]([a-zA-Z0-9-]*[a-zA-Z0-9])?$/g` (and a test to make sure the name is unique. In plain English: `Secret names may include letters, numbers, and dashes, and must not start or end with a dash.`

[the remark this answers is not in the document]
Comment (BenjaminMummery) [reply docs/tutorials/secrets.rst]: Thanks both, Rory I've taken the liberty of using your exact words XD

```
``name`` is the secret identifier.
Secret names may include letters, numbers, and dashes, and must not start or end with a dash.
This identifier is subsequently used to access the secret.
Secrets are scoped to a single user account, so there's no risk you're overriding somebody else's entry.
```

File: docs/tutorials/secrets.rst [hunk 2]
```diff
@@ -29,11 +37,16 @@ Then, you can read and set secrets within a script or a Python REPL:
     :dedent: 8
 
 
-``name`` is the secret identifier. It can be any string you like, as long as it fits in the length limit. This identifier is subsequently used to access the secret. Secrets are scoped to a single user account, so there's no risk you're overriding somebody else's entry.
+``name`` is the secret identifier.
+Secret names may include letters, numbers, and dashes, and must not start or end with a dash.
+This identifier is subsequently used to access the secret.
+Secrets are scoped to a single user account, so there's no risk you're overriding somebody else's entry.
```
Comment (roryschadler) [inline docs/tutorials/secrets.rst:43]: This isn't true, secrets are scoped to the workspace (so you could delete and recreate someone else's secret if you're in a shared workspace)

**Review by roryschadler — CHANGES_REQUESTED** [review]
Sorry, requested a quick change for something I noticed just now 😬

File: docs/tutorials/secrets.rst (hunk 2 above)
Comment (BenjaminMummery) [reply docs/tutorials/secrets.rst]: Good catch! New version now reads:

```
``name`` is the secret identifier.
Secret names may include letters, numbers, and dashes, and must not start or end with a dash.
This identifier is subsequently used to access the secret.
Secrets are scoped to a workspace, so care must be taken when managing secrets in shared workspaces.
```

**Review by roryschadler — APPROVED** [review]
Thanks!

File: docs/tutorials/secrets.rst (hunk 2 above)
Comment (roryschadler) [reply docs/tutorials/secrets.rst]: Thank you!
